blocks/editable/index.js

```diff
@@ -206,6 +206,14 @@ export default class Editable extends wp.element.Component {
 			event.preventDefault();
 			event.stopImmediatePropagation();
 		}
+
+		if ( event.keyCode === ENTER && event.shiftKey && this.props.inline ) {
```

Comment:
Can we include some more inline documentation describing what these various inline conditions are achieving? Seems like they'll be difficult to revisit in the future otherwise.